pennylane/ops/qubit/observables.py

```diff
@@ -435,13 +435,20 @@ class BasisStateProjector(Projector, Operation):
     # The call signature should be the same as Projector.__new__ for the positional
     # arguments, but with free key word arguments.
     def __init__(self, state, wires, id=None):
-        wires = Wires(wires)
-        state = list(qml.math.toarray(state).astype(int))
+        wires = qml.wires.Wires(wires)
 
-        if not set(state).issubset({0, 1}):
-            raise ValueError(f"Basis state must only consist of 0s and 1s; got {state}")
+        if state is None:
+            raise ValueError("Basis state cannot be None")
 
-        super().__init__(state, wires=wires, id=id)
+        if not qml.math.is_abstract(state):
+            state_array = qml.math.toarray(state)
+            state_array = state_array.astype(int)
+            if not qml.math.isininterval(state_array, 0, 1).all():
+                raise ValueError(f"Basis state must only consist of 0s and 1s; got {state_array}")
+        else:
+            state_array = qml.math.asarray(state)
```

Comment:
```suggestion
            # for cases like [jax.numpy.array(0), jax.numpy.array(1)]
            state_array = qml.math.asarray(state)
```

Might be worth adding a reminder for why we want to use `asarray` here instead of just letting `state` through unchanged.

Reply:
done, please review it and let me kno.